connector-api/src/main/java/it/pagopa/selfcare/mscore/api/UserConnector.java

```diff
@@ -50,13 +50,12 @@ public interface UserConnector {
 
     void findAndRemoveProduct(String userId, String institutionId, OnboardedProduct product);
 
-    List<OnboardedUser> findAllByIds(List<String> users);
+    List<OnboardedUser> findAllByIds(List<String> users, boolean existingOnly);
```

Comment:
What about introducing a new method `findExistingProductsByIds()`? 1. this usually results a better dx than polluting arguments with booleans 2. you avoid to refactor all exiting instaces of `findAllByIds`. You can keep all the existing code as a base for the two different methods making it private.

Reply:
Generally it's better to not expose public interfaces with boolean arguments. When really needed use an option object instead.

Reply:
you could introduce 2 method public (findAllByIds and indExistingProductsByIds) and using only one private method that implement the logic with boolean...